tensorboard/webapp/widgets/data_table/filter_dialog_test.ts

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {TestBed} from '@angular/core/testing';
+import {HarnessLoader} from '@angular/cdk/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {DataTableModule} from './data_table_module';
+import {FilterDialog} from './filter_dialog_component';
+import {DiscreteFilter, DomainType, IntervalFilter} from './types';
+import {By} from '@angular/platform-browser';
+import {RangeInputComponent} from '../range_input/range_input_component';
+import {RangeInputModule} from '../range_input/range_input_module';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import {MatCheckboxHarness} from '@angular/material/checkbox/testing';
+
+describe('filter dialog', () => {
+  let rootLoader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [FilterDialog],
+      imports: [DataTableModule, RangeInputModule, MatCheckboxModule],
+    }).compileComponents();
+  });
+
+  function createComponent(input: {filter: DiscreteFilter | IntervalFilter}) {
+    const fixture = TestBed.createComponent(FilterDialog);
+    fixture.componentInstance.filter = input.filter;
+    rootLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+    fixture.detectChanges();
+    return fixture;
+  }
+
+  it('dispatches event when an interval filter value is changed', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.INTERVAL,
+        includeUndefined: false,
+        minValue: 6,
```

Comment:
Can we add coverage of what is rendered? Given minValue, maxValue, filterLowerValue, filterUpperValue, what is actually passed to (or rendered by) the tb-range-component? (Ideally in its own test case).

Reply:
I've gone ahead and checked that the range input element is shown and that the initial values are correct. I don't want this test to overlap too greatly with the discrete range input test though. `tensorboard/webapp/widgets/range_input/range_input_test.ts`